src/cascadia/TerminalSettingsEditor/Actions.h

```diff
@@ -4,32 +4,85 @@
 #pragma once
 
 #include "Actions.g.h"
+#include "KeyBindingViewModel.g.h"
 #include "ActionsPageNavigationState.g.h"
+#include "RebindKeysEventArgs.g.h"
 #include "Utils.h"
+#include "ViewModelHelpers.h"
 
 namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
 {
-    struct CommandComparator
+    struct KeyBindingViewModelComparator
     {
-        bool operator()(const Model::Command& lhs, const Model::Command& rhs) const
+        bool operator()(const Editor::KeyBindingViewModel& lhs, const Editor::KeyBindingViewModel& rhs) const
         {
             return lhs.Name() < rhs.Name();
         }
     };
 
+    struct RebindKeysEventArgs : RebindKeysEventArgsT<RebindKeysEventArgs>
+    {
+    public:
+        RebindKeysEventArgs(const Control::KeyChord& oldKeys, const Control::KeyChord& newKeys) :
+            _OldKeys{ oldKeys },
+            _NewKeys{ newKeys } {}
+
+        WINRT_PROPERTY(Control::KeyChord, OldKeys, nullptr);
+        WINRT_PROPERTY(Control::KeyChord, NewKeys, nullptr);
+    };
+
+    struct KeyBindingViewModel : KeyBindingViewModelT<KeyBindingViewModel>, ViewModelHelper<KeyBindingViewModel>
+    {
+    public:
+        KeyBindingViewModel(const Control::KeyChord& keys, const Settings::Model::Command& cmd);
+
+        hstring Name() const { return _Command.Name(); }
+        hstring KeyChordText() const { return _KeyChordText; }
+        Settings::Model::Command Command() const { return _Command; };
+
+        // UIA Text
+        hstring EditButtonName() const noexcept;
+        hstring CancelButtonName() const noexcept;
+        hstring AcceptButtonName() const noexcept;
+        hstring DeleteButtonName() const noexcept;
+
+        void EnterHoverMode() { IsHovered(true); };

```

Comment:
I think you can actually bind function calls. `{x:Bind IsHovered(x:True)}`!

Reply:
You're not gonna believe this. That doesn't compile!! You get a syntax error in XamlTypeInfo.g.cpp because it's expecting the result of the function to be saved somewhere... but this is a void function.